### DISCUSSION TITLE
'sudo topgrade' is not working.

### DISCUSSION THREAD

**Discussion by lachsdachs** [topgrade-rs/topgrade#825] [Q&A] [answered]
i just tried to call `topgrade` with 'sudo` and `-y` so that i could run it in the background without intervention, but sudo didn't find the command. Confusingly, running 'sudo su' before running topgrade worked perfectly. I am somewhat new to linux, what did i mess up here?

**Comment by lucaspar** [comment, marked as answer]
can you paste the error you see when running with `sudo`?

I'm not sure if this is in the documentation somewhere, but my understanding is that we should avoid running `topgrade` as superuser directly. The tool will ask for your superuser password when required. If you want to avoid retyping it do as follows:

```bash
topgrade --edit-config
```

In the configuration file, add or uncomment the following line:

```toml
[misc]
    # ...
    pre_sudo = true
    # ...
```

---

Runing `topgrade` itself as superuser means running every upgraded tool as superuser - some of which might not be designed to be executed at this privilege level.

**Comment by lachsdachs** [comment]
The error message i receive is 'sudo: topgrade: command not found'.
Thank you for pointing out the config option, i will be using that instead.

**Comment by SteveLauC** [comment]
Yeah, you got the cmd not found error because sudo on Linux won't preserve PATH, see https://stackoverflow.com/questions/257616/why-does-sudo-change-the-path


And you should use the pre-sudo config entry.
